### ISSUE TITLE
[BUG/ISSUE] IndexError in GCPy 1-year transport tracers budget table routine

### ISSUE THREAD

**Issue by yantosca** [geoschem/gcpy#168]
I just ran across this issue in the 1-year TransportTracers benchmark code.  It happens in the `gcpy/budget_tt.py` routine, when it tries to save into the `q["Be7_f"]` numpy array:

```python
def annual_average_sources(globvars):
    """
    Computes the annual average of radionuclide sources.

    Args:
        globvars: obj of type _GlobVars
            Global variables needed for budget computations.

     Returns:
        result: dict
            Source totals in strat, trop, and strat+trop regimes.
    """

    # Initialize
    q = {}
    q_sum_f = np.zeros(globvars.N_MONTHS)
    q_sum_t = np.zeros(globvars.N_MONTHS)
    q_sum_s = np.zeros(globvars.N_MONTHS)
    result = {}

    # Pb210 source is from Rn222 decay, in the RadioNuclide collection
    q["Pb210_f"] = globvars.ds_dcy["PbFromRnDecay"].values \
        * globvars.kg_s_to_g_d["Pb210"]

    # Determine the shape of the data
    q_shape = q["Pb210_f"].shape     # e.g. (time,lev,lat,lon)
    n_levs = q_shape[1]              # Number of levels

    # Determine which array axes to sum over for monthly sums
    if globvars.is_gchp:
        area_var = "Met_AREAM2"
        sum_axes = (1, 2, 3, 4)
    else:
        area_var = "AREA"
        sum_axes = (1, 2, 3)

    # Be7 and Be10 sources are in the HEMCO diagnostics collection
    q["Be7_f"] = np.zeros(q_shape)
    q["Be10_f"] = np.zeros(q_shape)
    
    # Convert Be7 and Be10 sources from kg/m2/s to g/day
    # NOTE: This is a kludgey way to do it but it works and
    # preserves the shape of the data as (time,lev,lat,lon).
    for t in range(globvars.N_MONTHS):
        for k in range(n_levs):
            if globvars.is_gchp:
                q["Be7_f"][t, k, :, :, :] = \
                    globvars.ds_hco["EmisBe7_Cosmic"].isel(time=t, lev=k) * \
                    globvars.ds_met[area_var].isel(time=t) * \
                    globvars.kg_s_to_g_d["Be7"]
                q["Be10_f"][t, k, :, :, :] = \
                    globvars.ds_hco["EmisBe10_Cosmic"].isel(time=t, lev=k) * \
                    globvars.ds_met[area_var].isel(time=t) * \
                    globvars.kg_s_to_g_d["Be10"]
            else:
                q["Be7_f"][t, k, :, :] = \
                    globvars.ds_hco["EmisBe7_Cosmic"].isel(time=t, lev=k) * \
                    globvars.ds_met[area_var].isel(time=t) * \
                    globvars.kg_s_to_g_d["Be7"]
                q["Be10_f"][t, k, :, :] = \
                    globvars.ds_hco["EmisBe10_Cosmic"].isel(time=t, lev=k) * \
                    globvars.ds_met[area_var].isel(time=t) * \
                    globvars.kg_s_to_g_d["Be10"]
```
The error is:
```console
(gcpy) [holyjacob01 bmk]$ ./run_benchmark.py 1yr_tt_benchmark.yml 
Using configuration file 1yr_tt_benchmark.yml
The following plots and tables will be created for TransportTracersBenchmark:
 - Radionuclides budget table
 - Table of strat-trop exchange
 - Table of mass conservation
Comparisons will be made for the following combinations:
 - GCC vs GCC

%%% Creating GCC vs. GCC radionuclides budget table %%%
Using configuration file /n/holyscratch01/external_repos/GEOS-CHEM/gcgrid/geos-chem/validation/gcpy_test_data/1yr_transporttracer/GCC_dev/species_database.yml
Traceback (most recent call last):
  File "/n/holyscratch01/jacob_lab/ryantosca/bmk/./run_benchmark.py", line 1425, in <module>
    main()
  File "/n/holyscratch01/jacob_lab/ryantosca/bmk/./run_benchmark.py", line 1421, in main
    choose_benchmark_type(config)
  File "/n/holyscratch01/jacob_lab/ryantosca/bmk/./run_benchmark.py", line 99, in choose_benchmark_type
    run_1yr_tt_benchmark(
  File "/n/home09/ryantosca/python/gcpy/benchmark/modules/run_1yr_tt_benchmark.py", line 505, in run_be
nchmark
    ttbdg.transport_tracers_budgets(
  File "/n/home09/ryantosca/python/gcpy/gcpy/budget_tt.py", line 878, in transport_tracers_budgets
    data["src_total"] = annual_average_sources(globvars)
  File "/n/home09/ryantosca/python/gcpy/gcpy/budget_tt.py", line 568, in annual_average_sources
    globvars.ds_hco["EmisBe7_Cosmic"].isel(time=t, lev=k).values * \
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/xarray/core/dataarray.py",
 line 1130, in isel
    variable = self._variable.isel(indexers, missing_dims=missing_dims)
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/xarray/core/variable.py", 
line 1166, in isel
    return self[key]
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/xarray/core/variable.py", 
line 811, in __getitem__
    data = as_indexable(self._data)[indexer]
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/xarray/core/indexing.py", 
line 1362, in __getitem__
    return self.array[key.tuple]
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/dask/array/core.py", line 
1750, in __getitem__
    index2 = normalize_index(index, self.shape)
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/dask/array/slicing.py", li
ne 911, in normalize_index
    check_index(i, d)
  File "/n/home09/ryantosca/miniconda3/envs/gcpy/lib/python3.9/site-packages/dask/array/slicing.py", li
ne 981, in check_index
    raise IndexError(
IndexError: Index is not smaller than dimension 10 >= 10
```
This might not be a bug per se, but it may be behavior that used to work in an older versions of xarray/dask but does not work now. See this issue: https://github.com/dask/dask/issues/6468

I am using these library versions:
```console
xarray                        0.17.0
dask                          2021.7.1
numpy                         1.21.1
```
For now I will skip computation of the Rn-Pb-Be budget tables but it would be good to know how to resolve this issue.  I think we might have to rewrite the code in question.

@laestrada: any ideas?

**Comment by laestrada** [comment]
@yantosca at first glance it looks like the `globvars.ds_hco["EmisBe7_Cosmic"]` dataset is not the right shape to be selected by `.isel(time=t, lev=k)`. I would recommend saving `globvars.ds_hco["EmisBe7_Cosmic"]` to a file and troubleshooting it directly

**Comment by yantosca** [comment]
Ugh, there are some missing data files, so there are only 10 time points.  D'oh!

**Comment by yantosca** [comment]
Will try to put in an error trap here.

**Comment by yantosca** [comment]
This issue is now resolved with commit 93fdcca.
